File: zig-cli/src/ui/banner.zig
//! Banner UI component.

const std = @import("std");

/// Print the application banner.
pub fn print() void {
    const cyan = "\x1b[36m";
    const white_bold = "\x1b[1;37m";
    const dim = "\x1b[2m";
    const reset = "\x1b[0m";

    std.debug.print("{s}╔══════════════════════════════════════════════════════════╗{s}\n", .{ cyan, reset });
    std.debug.print("{s}║{s}{s}           📝 CONVEX NOTES MANAGER                        {s}{s}║{s}\n", .{ cyan, reset, white_bold, reset, cyan, reset });
    std.debug.print("{s}║{s}{s}         Self-Hosted • Zig Client • v0.1.0               {s}{s}║{s}\n", .{ cyan, reset, dim, reset, cyan, reset });
    std.debug.print("{s}╚══════════════════════════════════════════════════════════╝{s}\n\n", .{ cyan, reset });
}

File: go-cli/internal/ui/banner.go
// Package ui provides terminal UI components.
package ui

import (
	"fmt"

	"github.com/fatih/color"
)

// PrintBanner displays the application banner.
func PrintBanner() {
	cyan := color.New(color.FgCyan)
	white := color.New(color.FgWhite, color.Bold)
	dim := color.New(color.Faint)

	fmt.Println()
	cyan.Println("╔══════════════════════════════════════════════════════════╗")
	cyan.Print("║")
	white.Print("           📝 CONVEX NOTES MANAGER                        ")
	cyan.Println("║")
	cyan.Print("║")
	dim.Print("         Self-Hosted • Go Client • v0.1.0               ")
	cyan.Println("║")
	cyan.Println("╚══════════════════════════════════════════════════════════╝")
	fmt.Println()
}
